scripts/protocol-dts-generator.ts

```diff
@@ -3,37 +3,29 @@ import * as path from 'path'
 import {IProtocol, Protocol as P} from './protocol-schema'
 
 // TODO: @noj validate this via https://github.com/andischerer/typescript-json-typesafe against protocol-schema.d.ts
-import * as jsProtocol from '../json/js_protocol.json'
-import * as browserProtocol from '../json/browser_protocol.json'
-
-
-// TODO: @noj clean this up
-interface ClientDef {
-    name: string
-    client: P.FunctionType
-}
-
-type ParamOrFunction = P.ParameterType | P.FunctionType
+const jsProtocol: IProtocol = require('../json/js_protocol.json')
```

Comment:
Thank you 🙏 . If latest tsc does indeed validate the `require`-ed json against the IProtocol type, you can remove the TODO on line 5.

Reply:
>  If latest tsc does indeed validate the require-ed json against the IProtocol type, you can remove the TODO on line 5.

sadly it does not, the `require()` comes in as `any`, but if I do `"esModuleInterop": true` in tsconfig and go back to `import jsProtocol from '../json/js_protocol.json';` I can't apply an interface to the import statement (and if you apply the interface later, things like property `type` fields have been widened to `string` instead of their literal values so it fails).

Weirdly the js version of this works fine
```js
/** @typedef {import('./protocol-schema').IProtocol} IProtocol */

/** @type {IProtocol} */
const jsProtocol = require('../json/js_protocol.json')
```
and any changes to the schema or json are flagged as type errors.

So, leaving the TODO for now, but I can unassign you if you'd like :)

Reply:
works for me as is. Don't remove the TODO just yet.

I do quite like the work tsc team having doing regarding jsdoc type checking and `//@ts-check`. I know Google's lingua franca is Js. So if we want this file to be JS with jsdoc type annotations, I don't mind it at all.

Doesn't have to be this PR right away.